### ISSUE TITLE
Support NoSQL databases in generator

### ISSUE THREAD

**Issue by jessicard** [tinfoil/devise-two-factor#15]
It would be awesome if, in the generator, you could specify an optional arg that let you generate for a NoSQL database (like Mongoid). So instead of creating a database migration, it would just add the appropriate fields to the model file.

**Comment by JIucToyxuu** [comment]
I override some methods in my model and this work with mongoid. BTW: i use standard attr_encrypted(not fork).

`def encrypted_otp_secret=(value)
    self.write_attribute(:encrypted_otp_secret, value.force_encoding("ISO-8859-1").encode("UTF-8"))
  end`

`def encrypted_otp_secret
    if self.read_attribute(:encrypted_otp_secret).nil?
      super
    else
      self.read_attribute(:encrypted_otp_secret).encode("ISO-8859-1").force_encoding('ASCII-8BIT')
    end
  end`

**Comment by pabloq** [comment]
Is there a solution for this? It does not work with mongoid (5.1.6).... I tried everything from here, still receiving nil...

**Comment by abhipatnala** [comment]
@JIucToyxuu can you explain it bit more. probably with gist ?

**Comment by abhipatnala** [comment]
@QuinnWilton any update on this issue ?

**Comment by paul-mesnilgrente** [comment]
I have an error coming from `devise_two_factor/models/two_factor_authenticatable.rb:10`: `undefined method encrypts`

Do you know how to get this right?

**Comment by dcunning** [comment]
I've run into the same problem but cannot offer you a solution other than suggesting you stick to v4.1.1.

[The upgrade to v5.0.0](https://github.com/devise-two-factor/devise-two-factor/compare/v4.1.1...v5.0.0) made devise-two-factor rely on [active record's encrypts method](https://api.rubyonrails.org/classes/ActiveRecord/Encryption/EncryptableRecord.html#method-i-encrypts).  This was intentional as it's explicitly mentioned in [the repo's README](https://github.com/devise-two-factor/devise-two-factor?tab=readme-ov-file#getting-started) ("Devise-Two-Factor uses ActiveRecord encrypted attributes")

tl;dr __v5.0.0 adds activerecord as an undeclared dependency__, making it incompatible with any app that uses a NoSQL solution like mongoid over activerecord.
